File: src/main/java/mrfast/sbf/features/statDisplays/HealthDisplay.java
package mrfast.sbf.features.statDisplays;

import mrfast.sbf.SkyblockFeatures;
import mrfast.sbf.gui.components.Point;
import mrfast.sbf.gui.components.UIElement;
import mrfast.sbf.utils.GuiUtils;
import mrfast.sbf.utils.Utils;
import net.minecraft.client.Minecraft;


public class HealthDisplay {

    private static final Minecraft mc = Minecraft.getMinecraft();

    static {
        new HealthDisplayGUI();
    }

    static String display = Utils.health +"/"+Utils.maxHealth;
    public static class HealthDisplayGUI extends UIElement {
        public HealthDisplayGUI() {
            super("Health Display", new Point(0.39878336f, 0.8029036f));
            SkyblockFeatures.GUIMANAGER.registerElement(this);
        }

        @Override
        public void drawElement() {
            display = "§c"+Utils.nf.format(Utils.health)+"/"+Utils.nf.format(Utils.maxHealth);
            if(Utils.health >Utils.maxHealth) display="§6"+Utils.nf.format(Utils.health)+"§c/"+Utils.nf.format(Utils.maxHealth);

            GuiUtils.drawText(display, 0, 0, GuiUtils.TextStyle.BLACK_OUTLINE);
        }
        @Override
        public void drawElementExample() {
            display = "§c"+Utils.nf.format(Utils.health)+"/"+Utils.nf.format(Utils.maxHealth);
            if(Utils.health >Utils.maxHealth) display="§6"+Utils.nf.format(Utils.health)+"§c/"+Utils.nf.format(Utils.maxHealth);

            GuiUtils.drawText(display, 0, 0, GuiUtils.TextStyle.BLACK_OUTLINE);
        }

        @Override
        public boolean getToggled() {
            return SkyblockFeatures.config.HealthDisplay;
        }

        @Override
        public boolean getRequirement() {
            return Utils.inSkyblock;
        }
        @Override
        public int getHeight() {
            return Utils.GetMC().fontRendererObj.FONT_HEIGHT;
        }

        @Override
        public int getWidth() {
            return Utils.GetMC().fontRendererObj.getStringWidth(display);
        }
    }
}

File: src/main/java/mrfast/sbf/features/statDisplays/HealthDisplay.kt
package mrfast.sbf.features.statDisplays

import mrfast.sbf.SkyblockFeatures
import mrfast.sbf.gui.components.Point
import mrfast.sbf.gui.components.UIElement
import mrfast.sbf.utils.GuiUtils
import mrfast.sbf.utils.Utils

class HealthDisplay {
    var display: String = ""

    init {
        HealthDisplayGUI().register()
    }

    private fun updateDisplay() {
        display = buildString {
            append("§c").append(Utils.nf.format(Utils.health.toLong())).append("/").append(Utils.nf.format(Utils.maxHealth.toLong()))
            if (Utils.health > Utils.maxHealth) append("§6").append(Utils.nf.format(Utils.health.toLong())).append("§c/").append(Utils.nf.format(Utils.maxHealth.toLong()))
        }
    }

    inner class HealthDisplayGUI : UIElement("Health Display", Point(0.39878336f, 0.8029036f)) {
        override fun drawElement() {
            updateDisplay()
            GuiUtils.drawText(display, 0f, 0f, GuiUtils.TextStyle.BLACK_OUTLINE)
        }

        override fun drawElementExample() {
            drawElement();
        }

        override fun getToggled(): Boolean = SkyblockFeatures.config.HealthDisplay

        override fun getRequirement(): Boolean = Utils.inSkyblock

        override fun getHeight(): Int = Utils.GetMC().fontRendererObj.FONT_HEIGHT

        override fun getWidth(): Int = Utils.GetMC().fontRendererObj.getStringWidth(display)
    }
}
